nni/experiment/config/experiment_config.py

```diff
@@ -113,6 +113,11 @@ def _canonicalize(self, _parents):
             if algo is not None and algo.name == '_none_':
                 setattr(self, algo_type, None)
 
+        if self.advisor is not None:
+            assert self.tuner is None
```

Comment:
add a message here? for example, "advisor is not allowed when tuner is set."

Reply:
done